src/d3a/models/market/one_sided.py

```diff
@@ -199,7 +201,7 @@ def accept_offer(self, offer_or_id: Union[str, Offer], buyer: str, *, energy: in
         residual_offer = None
 
         if trade_rate is None:
-            trade_rate = offer.price / offer.energy
+            trade_rate = offer.energy_rate
```

Comment:
I guess could be used [here](https://github.com/gridsingularity/d3a/pull/675/files#diff-5c5fbb99e0155bbce18f8ca72347fd58R388-R389) too